tests/api.v2/events/handleWorkRequest.test.js

```diff
@@ -1,154 +1,39 @@
 const handleWorkRequest = require('../../../src/api.v2/events/handleWorkRequest');
-const CacheSingleton = require('../../../src/cache');
-const getPipelineStatus = require('../../../src/api.v2/helpers/pipeline/getPipelineStatus');
-const pipelineConstants = require('../../../src/api.v2/helpers/pipeline/constants');
-
-jest.mock('../../../src/api.v2/helpers/pipeline/getPipelineStatus');
-jest.mock('../../../src/cache');
-
-describe('handleWorkRequest', () => {
-  afterEach(() => {
-    jest.resetModules();
-    jest.restoreAllMocks();
-    jest.clearAllMocks();
-  });
-
-  it('Throws when an old timeout is encountered.', async () => {
-    // Initialize with an empty cache so a worker hit will be encountered.
-    CacheSingleton.createMock({});
-    expect.assertions(1);
-
-    const workRequest = {
-      ETag: '12345',
-      socketId: '6789',
-      experimentId: 'my-experiment',
-      timeout: '2001-01-01T00:00:00Z',
-      body: { name: 'GetEmbedding', type: 'umap', config: { minimumDistance: 0.3, distanceMetric: 'euclidean' } },
-    };
-
-    try {
-      await handleWorkRequest(workRequest);
-    } catch (e) {
-      expect(e.message).toMatch(
-        /^Request timed out at/,
-      );
-    }
-  });
-
-  it('Throws when type isnt valid.', async () => {
-    // Initialize with an empty cache so a worker hit will be encountered.
-    CacheSingleton.createMock({});
-    expect.assertions(1);
-
-    const workRequest = {
-      ETag: '12345',
-      socketId: '6789',
-      experimentId: 'my-experiment',
-      timeout: '2099-01-01T00:00:00Z',
-      body: { name: 'GetEmbedding', type: 'invalidType', config: { minimumDistance: 0.3, distanceMetric: 'invalidMetric' } },
-    };
-
-    try {
-      await handleWorkRequest(workRequest);
-    } catch (e) {
-      expect(e.message).toMatch(
-        /^Error: type does not match the pattern/,
-      );
-    }
-  });
-
-  it('Throws when an invalid distanceMetric in getEmbedding is encountered.', async () => {
-    // Initialize with an empty cache so a worker hit will be encountered.
-    CacheSingleton.createMock({});
-    expect.assertions(1);
-
-    const workRequest = {
-      ETag: '12345',
-      socketId: '6789',
-      experimentId: 'my-experiment',
-      timeout: '2099-01-01T00:00:00Z',
-      body: { name: 'GetEmbedding', type: 'umap', config: { minimumDistance: 0.3, distanceMetric: 'invalidMetric' } },
-    };
+const fake = require('../../test-utils/constants');
+
+jest.mock('../../../src/api.v2/middlewares/authMiddlewares');
+jest.mock('aws-xray-sdk');
+jest.mock('../../../src/api.v2/events/validateAndSubmitWork', () => (data) => {
+  if (data.uuid === 'invalid') {
+    throw new Error('invalid request');
+  }
+  return jest.fn();
+});
```

Comment:
Instead of doing this, can you do `mockValidateAndSubmitWork.mockImplementation(() => {})` in the test in which this case is tested?

Reply:
I can but this is generic for the whole test it is the same? Why would it be better to change it?